### DISCUSSION TITLE
Collapse TOC content in the margins in Book layour

### DISCUSSION THREAD

**Discussion by rressler** [quarto-dev/quarto-cli#2965] [Q&A]
Ref: Pushing the toc field all the way to the right, making document content wider #1461

Is there a way to allow one to collapse TOC in the left and/or right columns of HTML book outputs similar to code-fold?

I have HTML documents where display math content from the body overlaps  the TOC on the right.  If there were a code-fold like option to collapse it then I could just hide the TOC to see the output more clearly and then unhide it when needed.
Thanks

**Comment by dragonstyle** [comment]
A couple of thoughts:

- The sidebar and TOC should collapse automatically whenever it overlaps with content in the margin 
- We do support a 'reader-mode', which collapses sidebar and TOC which might suit you. More here: https://quarto.org/docs/websites/website-navigation.html#reader-mode

**Comment by rressler** [comment]
Charles,
Thanks for the quick response. I am viewing the HTML files in browser, not as part of a website. The sidebar and TOC do not collapse on display math and the reader-mode does not seem to work.
 In lieu of a reprex here is a picture.
Thanks,
Richard

[Graphical user interface, text, application  Description automatically generated]

From: Charles Teague ***@***.***>
Reply-To: quarto-dev/quarto-cli ***@***.***>
Date: Friday, October 21, 2022 at 10:37 AM
To: quarto-dev/quarto-cli ***@***.***>
Cc: Richard Ressler ***@***.***>, Author ***@***.***>
Subject: Re: [quarto-dev/quarto-cli] Collapse TOC content in the margins in Book layour (Discussion #2965)


External Email: Use caution with links and attachments.

A couple of thoughts:

  *   The sidebar and TOC should collapse automatically whenever it overlaps with content in the margin
  *   We do support a 'reader-mode', which collapses sidebar and TOC which might suit you. More here: https://quarto.org/docs/websites/website-navigation.html#reader-mode<https://urldefense.com/v3/__https:/quarto.org/docs/websites/website-navigation.html*reader-mode__;Iw!!IaT_gp1N!0K7Uo9b0MEzijMLovmphPR6A3oUc8_MG_kdl89zA5ajGeeGI446akskAv-9KXEDu0cTDXIHD5bsrhpyQ0GFwPgoVUcM$>

**Reply by dragonstyle** [reply]
Ah I see - you're right `reader-mode` is a website feature (not HTML). I'm guessing that perhaps the math isn't causing the TOC folding because we're not actually aware of the overlap (we evaluate elements whose explicit position extends into the margin regions vs detecting elements that are overflowing into the margin regions). If you can share a sample QMD I'm happy to see if there is something we could do about it, or if I have another suggestion (for example, perhaps you could wrap any overflowing math with a column class that lets our layout system know that it will flow into the margins).

**Comment by rressler** [comment]
Examples in sections 3 and 4

I am not sure I know how to do the column class suggestion. Would I create a new CSS class or use one of the existing ones.

The short term answer may be to rewrite and format the math so it does not overflow, to include any equation numbers.

Thanks,
Richard

From: Charles Teague ***@***.***>
Reply-To: quarto-dev/quarto-cli ***@***.***>
Date: Friday, October 21, 2022 at 12:15 PM
To: quarto-dev/quarto-cli ***@***.***>
Cc: Richard Ressler ***@***.***>, Author ***@***.***>
Subject: Re: [quarto-dev/quarto-cli] Collapse TOC content in the margins in Book layour (Discussion #2965)


External Email: Use caution with links and attachments.

Ah I see - you're right reader-mode is a website feature (not HTML). I'm guessing that perhaps the math isn't causing the TOC folding because we're not actually aware of the overlap (we evaluate elements whose explicit position extends into the margin regions vs detecting elements that are overflowing into the margin regions). If you can share a sample QMD I'm happy to see if there is something we could do about it, or if I have another suggestion (for example, perhaps you could wrap any overflowing math with a column class that lets our layout system know that it will flow into the margins).

**Reply by dragonstyle** [reply]
Sorry I'm having trouble seeing the examples - if you could share qmd source code that would be really helpful.

**Comment by rressler** [comment]
One was attached. Did it not come through?

From: Charles Teague ***@***.***>
Reply-To: quarto-dev/quarto-cli ***@***.***>
Date: Friday, October 21, 2022 at 4:41 PM
To: quarto-dev/quarto-cli ***@***.***>
Cc: Richard Ressler ***@***.***>, Author ***@***.***>
Subject: Re: [quarto-dev/quarto-cli] Collapse TOC content in the margins in Book layour (Discussion #2965)


External Email: Use caution with links and attachments.

Sorry I'm having trouble seeing the examples - if you could share qmd source code that would be really helpful.

**Reply by dragonstyle** [reply]
Yeah I don't think it did - email replies to GH discussion seem to often be problematic for attachments, screenshots, and so on.

**Comment by rressler** [comment]
[Archive.zip](https://github.com/quarto-dev/quarto-cli/files/9842744/Archive.zip)

**Reply by dragonstyle** [reply]
Ok give this a try - I just used fenced divs to give you a couple examples of extending the region (one across the whole page and another just to the right).

[Archive.zip](https://github.com/quarto-dev/quarto-cli/files/9843062/Archive.zip)

**Reply by rressler** [reply]
Thanks - This works. I now see what the documentation  (https://quarto.org/docs/authoring/article-layout.html#overflowing-content) means. Not only will the content overflow in a preferred direction (instead of by default to the right), but upon scrolling own and hitting the the  :::{.column-screen-inset-right} _overflow  math_ ::: , **it will collapse the TOC on the right**. If :::{.column-screen-inset-right}, the right margin TOC will collapse. If :::{.column-screen-inset-left}, the left TOC will collapse. If :::{.column-screen-inset}, both will collapse. You can click on the TOC to reveal itself, select an item, and then it collapses again (unless you select a different document in the left-hand TOC).

The TOC will remain collapsed no matter how you scroll after that so you can get by with only one in the document if appropriate.

When you overflow to the right the equation number will be all the way to the right margin so at times overflowing to the left might look nicer.
Thanks!
Richard

**Reply by dragonstyle** [reply]
Yes, exactly! One tweak to what you wrote- the TOC remaining collapsed after the first time it collapses is actually a bug that we have fixed in later releases. It should show itself once again when there is room between margin content and basically hide itself once again anytime it would overlap content in the margin.

**Comment by rressler** [comment]
Charles,
Even Better,
Thanks!
Richard

From: Charles Teague ***@***.***>
Reply-To: quarto-dev/quarto-cli ***@***.***>
Date: Monday, October 24, 2022 at 8:53 AM
To: quarto-dev/quarto-cli ***@***.***>
Cc: Richard Ressler ***@***.***>, Author ***@***.***>
Subject: Re: [quarto-dev/quarto-cli] Collapse TOC content in the margins in Book layour (Discussion #2965)


External Email: Use caution with links and attachments.

Yes, exactly! One tweak to what you wrote- the TOC remaining collapsed after the first time it collapses is actually a bug that we have fixed in later releases. It should show itself once again when there is room between margin content and basically hide itself once again anytime it would overlap content in the margin.
